src/app/containers/StakePage/components/FeeBlock.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useCallback, useMemo, useEffect } from 'react';
+import { useCachedAssetPrice } from 'app/hooks/trading/useCachedAssetPrice';
+import { useAccount } from 'app/hooks/useAccount';
+import { useTranslation } from 'react-i18next';
+import { AssetDetails } from 'utils/models/asset-details';
+import { getContract } from 'utils/blockchain/contract-helpers';
+import { Tooltip } from '@blueprintjs/core';
+import { numberFromWei, weiTo4 } from 'utils/blockchain/math-helpers';
+import {
+  staking_withdrawFee,
+  staking_numTokenCheckpoints,
+} from 'utils/blockchain/requests/staking';
+import { useStaking_getAccumulatedFees } from 'app/hooks/staking/useStaking_getAccumulatedFees';
+import { translations } from 'locales/i18n';
+import { LoadableValue } from 'app/components/LoadableValue';
+import { ContractName } from 'utils/types/contracts';
+import { bignumber } from 'mathjs';
+import { Asset } from 'types';
+import { weiToUSD } from 'utils/display-text/format';
+
+interface IFeeBlockProps {
+  contractToken: AssetDetails;
+  usdTotal: (asset: AssetDetails, value: number) => void;
+  useNewContract?: boolean;
+  title?: string;
+}
+
+export const FeeBlock: React.FC<IFeeBlockProps> = ({
+  contractToken,
+  usdTotal,
+  useNewContract = false,
+  title,
+}) => {
+  const account = useAccount();
+  const { t } = useTranslation();
+  const token =
+    contractToken.asset +
+    (contractToken.asset === Asset.SOV ? '_token' : '_lending');
+  const dollars = useCachedAssetPrice(contractToken.asset, Asset.USDT);
```

Comment:
Since you use `contractToken.asset` on multiple places and it's pretty verbose, extract it to a variable (`const {asset} = contractToken;`) and use the variable on all places.